src/pages/en/guides/deploy/google-cloud.mdx

```diff
@@ -9,32 +9,134 @@ i18nReady: true
+#### Create the Dockerfile
 
-5. Configure a port from which the files are served.
+##### Dockerfile for static sites (example)
 
-6. Enable public access by adding a new permission to `allUsers` called `Cloud Run Invoker`.
+This following Dockerfile will build your site and serve it using Apache HTTPD (default config) on port `80`.
 
-### Cloud Storage
+```docker title="Dockerfile"
+FROM node:lts as build
+WORKDIR /app
+COPY . .
+RUN npm i
+RUN npm run build
 
-1. Create a new GCP project, or select one you already have.
+FROM httpd:2.4 as runtime
+COPY --from=build /app/dist /usr/local/apache2/htdocs/
+EXPOSE 80
+```
 
-2. Create a new bucket under [Cloud Storage](https://cloud.google.com/storage).
+:::note[notes]
+If you're using a package manager other than `npm`, you'll need to change the commands:
 
-3. Give it a name and other required settings.
+```docker title="Dockerfile" add={4} "pnpm"
+FROM node:lts as build
+WORKDIR /app
+COPY . .
+RUN npm i -g pnpm@latest
+RUN pnpm i
+RUN pnpm run build
+...
+```
+:::
 
-4. Upload your `dist` folder into it or upload using [Cloud Build](https://cloud.google.com/build).
+##### Dockerfile for SSR sites (example)
 
-5. Enable public access by adding a new permission to `allUsers` called `Storage Object Viewer`.
+This following Dockerfile will build your site and serve it using Node.js (SSR adapter) on port `3000`.
+
+```docker title="Dockerfile"
+FROM node:lts as runtime
+WORKDIR /app
+
+COPY . .
+
+RUN npm install
+RUN npm run build
+
+ENV HOST=0.0.0.0
+ENV PORT=3000
+EXPOSE 3000
+CMD node ./dist/server/entry.mjs
+```
+
+:::tip[Keep this in mind]
+These are just examples of Dockerfiles. You can customize them to your needs. For example, you could use another image, like `node:lts-alpine`.
+
+```docker title=".Dockerfile del={1} add={2}
+FROM node:lts as runtime
+FROM node:lts-alpine as runtime
+```
+:::
+
+#### Build the Container
+
+Once the Dockerfile is created, build it into an image and push it to Google Cloud. There are a few ways to accomplish this:
+
+**Build locally using Docker**:
+
+Use the `docker build` command to build the image, `docker tag` to give it a tag, then `docker push` to push it to a registry. In the case of Google Cloud, [`Google Cloud Registry`](https://cloud.google.com/container-registry/docs/pushing-and-pulling#add-registry) is the easiest option, but you can also use [Docker Hub](<https://hub.docker.com/>).
+
+```bash
+# build your container
+docker build .
+
+docker tag SOURCE_IMAGE HOSTNAME/PROJECT-ID/TARGET-IMAGE:TAG
+
+# Push your image to a registry
+docker push HOSTNAME/PROJECT-ID/IMAGE:TAG
+```
+
+:::note[Replace variables]
+Replace the following:
+
+`SOURCE_IMAGE` is the local image name or image ID.
+`HOSTNAME` is the registry host (`gcr.io`, `eu.grc.io`, `asia.gcr.io`, `us.gcr.io`).
+`PROJECT` is your Google Cloud project ID.
+`TARGET-IMAGE` is the name for the image when it's stored in the registry.
+`TAG` is the version associated with the image
+:::
+
```

Comment:
````suggestion
Use the following values from your own project in your commands:

- `SOURCE_IMAGE`: the local image name or image ID.

- `HOSTNAME`: the registry host (`gcr.io`, `eu.grc.io`, `asia.gcr.io`, `us.gcr.io`).

- `PROJECT`: your Google Cloud project ID.

- `TARGET-IMAGE`: the name for the image when it's stored in the registry.

- `TAG` is the version associated with the image

````
Removing the note and using bullet points for readability! Not entirely sold on the intro phrase, but I feel like it should refer to the commands somehow.

Reply:
What do you think of this intro phrase? 

**"Change the following values in the commands above to match your project:"**

Reply:
I'm happy with that!